config/src/main/java/org/springframework/security/config/annotation/web/configurers/saml2/Saml2LoginConfigurer.java

```diff
@@ -103,10 +106,14 @@
 
 	private RelyingPartyRegistrationRepository relyingPartyRegistrationRepository;
 
+	private Saml2MetadataResolver saml2MetadataResolver;
+
 	private AuthenticationManager authenticationManager;
 
 	private Saml2WebSsoAuthenticationFilter saml2WebSsoAuthenticationFilter;
 
+	private Saml2MetadataFilter saml2MetadataFilter;
```

Comment:
Entity ID patterns used as defaults across spring-security-saml map to the URL, which for me suggest this URL should be available without any external configuration. Why do you think it's odd?

Reply:
> Why do you think it's odd?

Because `saml2Login()` indicates an authentication mechanism, so it should only contain filters related to that. 

For example, `saml2Login()` isn't the place to configure SAML logout.

> without any external configuration

Can you elaborate on this point? What I'm suggesting is that an application do:

```java
http
    .saml2Login(saml2 -> {})
    .addFilter(new Saml2MetadataFilter(new OpenSamlMetadataResolver()));
```

which already seems simple.